File: data/src/main/java/home/oleg/placesnearme/model/DetailedVenueWithPhotos.java
package home.oleg.placesnearme.model;

import android.arch.persistence.room.Embedded;
import android.arch.persistence.room.Relation;

import java.util.List;


public class DetailedVenueWithPhotos {

    @Embedded
    private DetailedVenueDbEntity venue;

    @Relation(parentColumn = "id", entityColumn = "venueId")
    private List<PhotoDbEntity> photos;

    public DetailedVenueDbEntity getVenue() {
        return venue;
    }

    public void setVenue(DetailedVenueDbEntity venue) {
        this.venue = venue;
    }

    public List<PhotoDbEntity> getPhotos() {
        return photos;
    }

    public void setPhotos(List<PhotoDbEntity> photos) {
        this.photos = photos;
    }
}

File: core-data/src/main/java/home/oleg/placesnearme/coredata/model/DetailedVenueWithPhotos.kt
package home.oleg.placesnearme.coredata.model

import androidx.room.Embedded
import androidx.room.Relation

data class DetailedVenueWithPhotos(
        @Embedded
        var venue: DetailedVenueDbEntity) {

    @Relation(parentColumn = "id", entityColumn = "venueId")
    var photos: List<PhotoDbEntity> = emptyList()
}

